server/handlers/handlers.go

```diff
@@ -127,8 +128,15 @@ func (ah *ActivityHandler) Stop() {
 }
 
 func (ah *ActivityHandler) Handle(activity msteams.Activity) error {
-	ah.queue <- activity
 	metrics := ah.plugin.GetMetrics()
+	if len(ah.queue) >= activityQueueSize {
+		if metrics != nil {
```

Comment:
We added it at various places because we were getting panics on the startup of the plugin due to this [line](https://github.com/mattermost/mattermost-plugin-msteams-sync/blob/17092cd65bdfc29f9d3240aaa6251d0bbc026b47/server/handlers/handlers.go#L92) or some other lines where we access the metrics, as the function was returning nil when we disable the `Enable Performance Monitoring` in the system console.

Reply:
Ah, maybe this was a result of moving `GetMetrics()` from returning a concrete implementation to an interface. Normally, Go doesn't panic on a `nil` receiver, but indeed if the interface is `nil`, the panic would occur. Let me brew on this a bit, since it's painful (and risky to remember) to check this everywhere.

Reply:
This was bugging me enough that I've submitted https://github.com/mattermost/mattermost-plugin-msteams-sync/pull/380 for discussion :)